jena-fuseki2/jena-fuseki-embedded/src/test/java/org/apache/jena/fuseki/embedded/FusekiTestAuth.java

```diff
@@ -0,0 +1,182 @@
+ *     SecurityHandler sh = FusekiAuth.makeSimpleSecurityHandler("/*", "USER", "PASSWORD");
+ *     FusekiAuth.setupServer(true, sh);
+ * }
+ * 
+ * &#64;AfterClass
+ * public static void afterClassAuth() {
+ *     FusekiAuth.teardownServer();
+ *     // Clear up any pooled connections.
+ *     HttpOp.setDefaultHttpClient(HttpOp.createPoolingHttpClient());
+ * }
+ * 
+ * &#64;Test
+ * public void myAuthTest() {
+ *     BasicCredentialsProvider credsProvider = new BasicCredentialsProvider();
+ *     Credentials credentials = new UsernamePasswordCredentials("USER", "PASSWORD");
+ *     credsProvider.setCredentials(AuthScope.ANY, credentials);
+ *     HttpClient client = HttpClients.custom().setDefaultCredentialsProvider(credsProvider).build();
+ *     try (TypedInputStream in = HttpOp.execHttpGet(ServerCtl.urlDataset(), "* /*", client, null)) {}
+ * }
+ * 
+ * &#64;Test
+ * public void myAuthTestRejected() {
+ *     BasicCredentialsProvider credsProvider = new BasicCredentialsProvider();
+ *     Credentials credentials = new UsernamePasswordCredentials("USER", "PASSWORD");
+ *     credsProvider.setCredentials(AuthScope.ANY, credentials);
+ *     HttpClient client = HttpClients.custom().setDefaultCredentialsProvider(credsProvider).build();
+ *     try (TypedInputStream in = HttpOp.execHttpGet(ServerCtl.urlDataset(), "* /*", client, null)) {}
+ *     catch (HttpException ex) {
+ *         throw assertAuthHttpException(ex);
+ *     }
+ * }
+ * </pre>
+ * 
+ * {@code @BeforeClass} can be {@code @Before} but server stop-start is expensive so a
+ * large test suite may end up quite slow.
+ */
+public class FusekiTestAuth {
+    private static int currentPort = choosePort() ;
+    
+    public static int port() {
+        return currentPort ;
+    }
+    
+    static boolean CLEAR_DSG_DIRECTLY = true ;
+    static private DatasetGraph dsgTesting ;
+    
+    // Abstraction that runs a SPARQL server for tests.
+    public static final String urlRoot()            { return "http://localhost:"+port()+"/" ; }
+    public static final String datasetPath()        { return "/dataset" ; }
+    public static final String urlDataset()         { return "http://localhost:"+port()+datasetPath() ; }
+    public static final DatasetGraph getDataset()   { return dsgTesting ; }
+    
+    public static final String serviceUpdate()      { return "http://localhost:"+port()+datasetPath()+"/update" ; } 
+    public static final String serviceQuery()       { return "http://localhost:"+port()+datasetPath()+"/query" ; }
+    public static final String serviceGSP()         { return "http://localhost:"+port()+datasetPath()+"/data" ; }
+    
+    private static FusekiEmbeddedServer server ;
+    
+    public static void setupServer(boolean updateable, SecurityHandler sh) {
+        // Clean datasets.
+        dsgTesting = DatasetGraphFactory.createTxnMem();
+        server = FusekiEmbeddedServer.create()
+            .add(datasetPath(), dsgTesting)
```

Comment:
Speed but also for clear-up.  No files get left on disk to interfere with other tests; especially if the server takes awhile to stop properly and release all its resources.

We could add `void setupServer(boolean updateable, DatasetGraph dsg, SecurityHandler sh)` for the general case.

`FusekiTestServer` is also a in-memory transactional dataset and there it is much more difficult to inject a specific implementation because the lifecycle is driven by JUnit.

The example on the JIRA shows it is not necessary to use these helper classes.  They are helpful when a lot of small tests are run, ensuring a balance between careful clean-up and speed.